pkg/ctl/cmdutils/nodegroup.go

```diff
@@ -21,33 +21,40 @@ func PopulateNodegroup(stackManager manager.StackManager, name string, cfg *api.
 		}
 		nodeGroupType = api.NodeGroupTypeUnowned
 	}
+	if err = PopulateNodegroupFromStack(nodeGroupType, name, cfg); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PopulateNodegroupFromStack populates the nodegroup field of an api.ClusterConfig by type from its CF stack.
+func PopulateNodegroupFromStack(nodeGroupType api.NodeGroupType, nodeGroupName string, cfg *api.ClusterConfig) error {
 	switch nodeGroupType {
 	case api.NodeGroupTypeUnmanaged:
-		cfg.NodeGroups = []*api.NodeGroup{
-			{
-				NodeGroupBase: &api.NodeGroupBase{
-					Name: name,
-				},
-			},
-		}
+		PopulateUnmanagedNodegroup(nodeGroupName, cfg)
 	case api.NodeGroupTypeManaged:
-		cfg.ManagedNodeGroups = []*api.ManagedNodeGroup{
-			{
-				NodeGroupBase: &api.NodeGroupBase{
-					Name: name,
-				},
+		cfg.ManagedNodeGroups = append(cfg.ManagedNodeGroups, &api.ManagedNodeGroup{
+			NodeGroupBase: &api.NodeGroupBase{
+				Name: nodeGroupName,
 			},
-		}
+		})
 	case api.NodeGroupTypeUnowned:
-		cfg.ManagedNodeGroups = []*api.ManagedNodeGroup{
-			{
-				NodeGroupBase: &api.NodeGroupBase{
-					Name: name,
-				},
-				Unowned: true,
+		cfg.ManagedNodeGroups = append(cfg.ManagedNodeGroups, &api.ManagedNodeGroup{
+			NodeGroupBase: &api.NodeGroupBase{
+				Name: nodeGroupName,
 			},
-		}
+			Unowned: true,
+		})
 	}
```

Comment:
what happens if I'm deleting via a cfg file? e.g.:
`eksctl delete cluster -f cluster.yaml`

and that `cluster.yaml` already contains (all, or some) of the nodegroups? Won't this result in the in-memory cluster config having duplicate entries for the same nodegroup? Might need to check if the nodegroup is already in the cfg before adding

Reply:
thank you, missed that! I added code for when there is a cfg file and tested it manually :)

Reply:
https://github.com/weaveworks/eksctl/pull/4205/commits/96eec779f4ae575d0b449995bbe22cc9b97fc7c8

Reply:
mhmm this introduces new problems:
1. config:
```
nodegroups:
- name: ng-1
- name: ng-2
```
2. eksctl create nodegroup -f cluster.yaml
3. create a new nodegroup imperatively, e.g. `eksctl create nodegroup --name ng-3`
4. eksctl delete cluster -f cluster.yaml

we would only attempt to delete `ng-1` and `ng-2`, meaning we would fail when deleting the cluster stack later as `ng-3` still exists.

Reply:
We can either:
- ignore `cfg.Nodgroups` and just use the stacks as the source of truth
- Merge the existing `cfg.Nodegroups` and the nodegroup stacks that aren't in the `cfg.nodegroups` together into the `cfg.Nodegroups`

Reply:
edited my commit to https://github.com/weaveworks/eksctl/pull/4205/commits/11805e8976a5cfd06a839ea4864f6627c308bf01 to use the stacks as the source of truth!

makes sense since, if they're deleting a cluster, they just want everything gone